internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,293 @@
+	Kubeclient kubernetes.Interface
+
+	// GRPCPort is the container grpc port which is defaulted to 50051
+	GRPCPort int32
+}
+
+// newRegistryPod initializes the RegistryPod struct and sets defaults for empty fields
+func newRegistryPod(kubeclient kubernetes.Interface, dbPath, bundleImage, namespace string) *RegistryPod {
+	rp := &RegistryPod{}
+
+	if rp.GRPCPort == 0 {
+		rp.GRPCPort = defaultGRPCPort
+	}
+
+	if rp.IndexImage == "" {
+		rp.IndexImage = defaultIndexImage
+	}
+
+	if rp.BundleAddMode == "" {
+		if rp.IndexImage == defaultIndexImage {
+			rp.BundleAddMode = semverBundleAddMode
+		} else {
+			rp.BundleAddMode = replacesBundleAddMode
+		}
+	}
+	rp.Kubeclient = kubeclient
+	rp.DBPath = dbPath
+	rp.BundleImage = bundleImage
+	rp.Namespace = namespace
+
+	return rp
+}
+
+// CreateOnCluster returns a bundle registry pod built from an index image
+// after verifying successful creation of the pod, or returns the pod logs and error in case of failures
+func (rp *RegistryPod) CreateOnCluster(ctx context.Context) (*corev1.Pod, string, error) {
+	// validate the RegistryPod struct and ensure required fields are set
+	err := rp.validate()
+	if err != nil {
+		return nil, "", fmt.Errorf("error in validating RegistryPod struct: %v", err)
+	}
+
+	// create the registry pod on the cluster
+	pod, err := rp.createPodOnCluster(ctx)
+	if err != nil {
+		podLogs, logErr := rp.FetchPodLogs(ctx, pod)
+		if logErr != nil {
+			return nil, podLogs, fmt.Errorf("error creating pod: %v: and fetching logs: %v", err, logErr)
+		}
+		return nil, podLogs, fmt.Errorf("error creating pod: %v", err)
+	}
+
+	// verify that pod is successfully created
+	err = rp.validatePodOnCluster(ctx, pod)
+	if err != nil {
+		podLogs, logErr := rp.FetchPodLogs(ctx, pod)
+		if logErr != nil {
+			return nil, podLogs, fmt.Errorf("error verifying pod creation: %v: and fetching logs: %v", err, logErr)
+		}
+		return nil, podLogs, fmt.Errorf("error verifying pod creation: %v", err)
+	}
+
+	return pod, "", nil
+}
+
+// createPodOnCluster returns a pod built from an index image or returns if an existing pod with the same name is found
+func (rp *RegistryPod) createPodOnCluster(ctx context.Context) (*corev1.Pod, error) {
+	// call podForBundleRegistry() to make the pod definition
+	pod, err := rp.podForBundleRegistry()
+	if err != nil {
+		return nil, fmt.Errorf("error in building registry pod: %v", err)
+	}
+
+	// Check if Pod already exists
+	tmpPod, err := rp.Kubeclient.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			// Create pod in kubernetes cluster using the above pod definition and clientset
+			pod, err = rp.Kubeclient.CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{})
+			if err != nil {
+				return nil, fmt.Errorf("error creating registry pod: %v", err)
+			}
+		} else {
+			return nil, fmt.Errorf("error getting existing registry pod: %v", err)
+		}
```

Comment:
No need to else: (go simple)
```suggestion
		} 
		return nil, fmt.Errorf("error getting existing registry pod: %v", err)
```